connect/runtime/src/test/java/org/apache/kafka/connect/integration/MonitorableSinkConnector.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.integration;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.Task;
+import org.apache.kafka.connect.runtime.TestSinkConnector;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.kafka.connect.sink.SinkTask;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class MonitorableSinkConnector extends TestSinkConnector {
+
+    private static final Logger log = LoggerFactory.getLogger(MonitorableSinkConnector.class);
+
+    public static final AtomicInteger COUNTER = new AtomicInteger();
```

Comment:
I think `CountDownLatch` would be preferable so that we don't call `sleep` in any place in the code. 

Feel free to declare more than one if you have more than one waiting conditions.

Reply:
it's not too straightforward putting a Latch here since those can't be reset, and we would have to recreate those for every test.

Reply:
And why not allocating a new one in every test?
I really believe that `sleep` eventually leads to flaky tests and also introduces unneeded, constant delays, that eventually add up and make tests unbearable to sit and wait for to finish.